File: Golang/hashtable/q1/q1.go
package q1

type Solution1 struct{}

func (s *Solution1) TwoSum(nums []int, target int) []int {
	hashtable := map[int]int{}
	for index, val := range nums {
		if findIndex, ok := hashtable[target-val]; ok {
			return []int{findIndex, index}
		}
		hashtable[val] = index
	}
	return nil
}

File: Csharp/Hashtable/Q1/Q1.cs
namespace Csharp.Hashtable.Q1;

public class Solution1
{
    public int[] TwoSum(int[] nums, int target)
    {
        var hashtable = new Dictionary<int, int>();
        for (var i = 0; i < nums.Length; i++)
        {
            if (hashtable.ContainsKey(target - nums[i]))
            {
                return new[] { hashtable[target - nums[i]], i };
            }
            hashtable[nums[i]] = i;
        }
        return new[] { -1, -1 };
    }
}
